File: EtiveMor.OpenImoCiiCalculator/EtiveMor.OpenImoCiiCalculator.Core/Services/Impl/ShipTransportWorkCalculatorService.cs
namespace EtiveMor.OpenImoCiiCalculator.Core.Services.Impl
{
    public class ShipTransportWorkCalculatorService : IShipTransportWorkCalculatorService
    {

        /// <summary>
        /// Calculates a ship's transport work, which is the product of the ship's capacity and the distance sailed in a calendar year
        /// </summary>
        /// <param name="capacity">
        /// The ship's capacity, as calculated in accordance with MEPC 337(76)
        /// </param>
        /// <param name="distanceSailed">
        /// The distance sailed in nautical miles (nm) in a calendar year
        /// </param>
        /// <returns>
        /// A double representing the ship's transport work in a calendar year
        /// </returns>
        /// <remarks>
        /// A ship's capacity can be calculated with <seealso cref="ShipCapacityCalculatorService.GetShipCapacity(Models.Enums.ShipType, double, double)"/>
        /// </remarks>
        public double GetShipTransportWork(double capacity, double distanceSailed)
        {
            return capacity * distanceSailed;
        }
    }
}

File: python-impl/open_imo_cii_calculator/services/ship_transport_work_calculator_service.py
"""
Service for calculating ship transport work
"""


class ShipTransportWorkCalculatorService:
    """
    Service for calculating the transport work of a ship
    """
    
    def get_ship_transport_work(self, capacity: float, distance_travelled_in_nautical_miles: float) -> float:
        """
        Gets a ship's transport work, which is the product of the ship's capacity and the distance travelled
        in a calendar year expressed in nautical miles.

        Args:
            - capacity (float): The ship's capacity (pre-calculated)
            - distance_travelled_in_nautical_miles (float): The distance travelled by the ship in a calendar year in nautical miles

        Returns:
            - float: The ship's transport work
        """
        if capacity <= 0:
            raise ValueError("Capacity must be a positive value")
            
        if distance_travelled_in_nautical_miles <= 0:
            raise ValueError("Distance travelled must be a positive value")
            
        return capacity * distance_travelled_in_nautical_miles